File: app/src/main/java/com/bresiu/codechallenge/repository/mappers/PostModelToEntityMapper.java
package com.bresiu.codechallenge.repository.mappers;

import com.bresiu.codechallenge.data.entity.Post;

public class PostModelToEntityMapper
		implements Mapper.ObjectMapper<Post, com.bresiu.codechallenge.model.Post> {
	@Override public Post map(com.bresiu.codechallenge.model.Post modelPost) {
		return new Post(modelPost.id, modelPost.userId, modelPost.title, modelPost.body);
	}
}

File: app/src/main/java/com/bresiu/codechallenge/repository/mappers/PostModelToEntityMapper.kt
package com.bresiu.codechallenge.repository.mappers

import com.bresiu.codechallenge.data.entity.Post

class PostModelToEntityMapper : Mapper.ObjectMapper<Post, com.bresiu.codechallenge.model.Post> {
  override fun map(mappingItem: com.bresiu.codechallenge.model.Post): Post {
    return Post(mappingItem.id.toLong(), mappingItem.userId.toLong(), mappingItem.title, mappingItem.body)
  }
}